features/appmgt/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/apis/v1/lifecycle_api_router.jag

```diff
@@ -131,364 +131,405 @@ require('/modules/publisher.js').exec(function (ctx) {
+                content.messages.push("You do not have permissions to perform action '"
+                                      + action + "' on version "
+                                      + currentArtifact.attributes.overview_version
+                                      + " of '" + currentArtifact.attributes.overview_name + "' "
+                                      + appType);
+                response.content = stringify(content);
+                return;
+            }
+        } catch (e) {
+            // this exception is thrown by
+            // org.wso2.carbon.user.core.AuthorizationManager.isRoleAuthorized(roleName,
+            // resourceId, action) throws UserStoreException
+            log.error("Cannot retrieve permissions for action '" + action + "' on artifact '" +
+                      currentArtifact.path + "' for user " + user.username);
+            log.error(e);
+            response.status = 500;
+            response.contentType = "application/json";
+            content.status = "Error";
+            content.messages.push("Cannot retrieve permissions for action '" + action + "'");
+            response.content = stringify(content);
+            return;
+        }
+
+        /**
+         * If everything went correctly in previous executions, previousArtifacts array should
+         * contain only 01 item. However due to some failures, previousArtifacts can contain
+         * multiple items. That means there are more than 1 published version of this app.
+         * In order to handle both situations, previousArtifacts array is processed as a
+         * multi-item array in the following code.
+         * @type {Object[]}
+         */
+        var previousArtifacts = [];
+        // retrieve previously published web app versions (same provider & name)
+        try {
+            previousArtifacts = artifactManager.find(function (artifact) {
+                var provider = (artifact.attributes.overview_provider ==
+                                currentArtifact.attributes.overview_provider);
+                var name = (artifact.attributes.overview_name ==
+                            currentArtifact.attributes.overview_name);
+                return (provider && name && (artifact.lifecycleState == 'Published'));
+            }, null);
+        } catch (e) {
+            log.error("Cannot retrieve previously published " + appType + " "
+                      + currentArtifact.attributes.overview_name + " by "
+                      + currentArtifact.attributes.overview_provider);
+            log.error(e);
+            // continue work flow
+        }
+
+        /** @type {{isSuccess: boolean, message: string}} */
+        var operationResult;
+
+        // For all previously published apps when publishing or re-publishing current app,
+        // unpublish previously published version of this app.
+        if (((action == "Publish") || (action == "Re-Publish"))
```

Comment:
Please fixed this file code reviews as commented in https://github.com/wso2/product-app-manager/pull/234